service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -211,6 +210,24 @@ where
     writeln!(writer, "}}")
 }
 
+pub fn get_pagination_item_type(
+    pagination: &Pagination,
+    service: &Service,
+    operation: &Operation,
+    for_timestamps: &str
+) -> Option<String> {
+    service.get_shape(operation.output.as_ref().map(|o| o.shape.as_str()).expect(&format!("{} operation {} output did not contain a shape for paginated method", service.name(), operation.name)))
```

Comment:
I realize this is deep so I feel obligated to explain

The assumption here is that for paginatable types, we have an output shape which contains a member that is of type Vec<SomeThing> and I want to gain access to the name of `Something` for use in `impl Stream<Item=Some, ...>`. 

Botocor's paginator definition gives me the name of the member as a result key but not it's shape, hence we go digging.

Reply:
Cool, that makes sense to me! I feel like it would be valuable to actually just add that comment into the code for future readers?

Reply:
Looking at all the `and_then`'s below, do you know that you can use `?` in an `Option` context? That might help make the code a bit more straight-forward.